e2etests/test/twin_disconnect.js

```diff
@@ -95,175 +94,235 @@ var protocolAndTermination = [
   },
 ];
 
-
-var newProps = {
-  foo : 1,
-  bar : {
-    baz : 2,
-    tweedle : {
-      dee : 3
-    }
-  }
-};
-
-var moreNewProps = {
-  bar : {
-    baz : 3
-  }
-};
-
-var nullMergeResult = JSON.parse(JSON.stringify(newProps));
-delete nullMergeResult.tweedle;
-
 protocolAndTermination.forEach( function (testConfiguration) {
   describe(testConfiguration.transport.name + ' using device/service clients - disconnect twin', function () {
 
     this.timeout(60000);
-    var deviceClient, deviceTwin;
-    var serviceTwin;
-
-    var deviceDescription;
+    // test clients
+    var deviceClient, deviceTwin, serviceTwin;
 
-    beforeEach(function (done) {
-      setTwinMoreNewPropsTimeout = null;
-      var host = ConnectionString.parse(hubConnectionString).HostName;
-      var pkey = new Buffer(uuid.v4()).toString('base64');
-      var deviceId = '0000e2etest-delete-me-twin-e2e-disconnect-' + uuid.v4();
+    // timeouts that could trigger test callbacks
+    var twinUpdateAfterFaultInjectionTimeout, faultInjectionTimeout;
 
-      deviceDescription = {
-        deviceId:  deviceId,
-        status: 'enabled',
-          authentication: {
-          symmetricKey: {
-            primaryKey: pkey,
-            secondaryKey: new Buffer(uuid.v4()).toString('base64')
-          }
-        },
-        connectionString: 'HostName=' + host + ';DeviceId=' + deviceId + ';SharedAccessKey=' + pkey
-      };
+    var deviceDescription;
+    var registry = Registry.fromConnectionString(hubConnectionString);
 
-      var registry = Registry.fromConnectionString(hubConnectionString);
+    before(function (beforeCallback) {
+      debug('creating test device...');
+      DeviceIdentityHelper.createDeviceWithSas(function (err, testDeviceInfo) {
+        if (err) {
+          debug('failed to create test device: ' + err.toString());
```

Comment:
>err.toString() [](start = 51, length = 14)

Reasonable to add deviceId? other places in the before/after/each?

Reply:
at this point we don't have a device id. only an error, that's because the device id is created by the DeviceIdentityHelper itself. I'm adding logs to DeviceIdentityHelper.